client/blocks/README.md

```diff
@@ -0,0 +1,39 @@
+# Why we chose to use our pre-existing AJAX APIs instead of the WooCommerce Blocks APIs for Payment Request Buttons
+
+While we were in the process of migrating the Blocks integration from the WooCommerce Blocks plugin to Stripe we encountered several issues while adding support for Payment Requests with the Blocks API. The easiest way to fix these issues at the time was to use our pre-existing AJAX APIs and forego the Blocks API entirely.
+
+The WooCommerce Blocks API has since been significantly improved and we need to re-evaluate whether we can now use the Blocks API, see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1573.
+
+This file is meant to document why we chose to forego the Blocks API at the time, and what the downsides and benefits of that approach are.
+
+## What were the issues we encountered?
+
+-   3DS payments didn't work with saved cards and Payment Requests because some scripts weren't loaded in those contexts.
+-   The Payment Request Dialog [didn't close during 3DS validation](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1467#discussion_r623877272).
+
+Both of these have since been addressed in WooCommerce Blocks.
+
+## The Downsides of Using the AJAX API
+
+We're unable to use the framework provided by WooCommerce Blocks during checkout with Payment Request Buttons, and we miss out on many things Blocks provides by default, such as a responsive UI, improved error reporting, and more.
+
+More specifically:
+
+-   We have to manually make the UI responsive, and some parts we simply _can't_ make responsive.
+    -   This refers to things like blocking the UI during the payment process, starting and stopping loading indicators, and general UX improvements in that vein.
```

Comment:
By "responsive" here, I ended up interpreting it as responsive in the sense of being adjustable to different screen sizes. But that's not the case, right?

Actually, the blocking of the UI and the loading indicators were missing from the Blocks implementation too. Most of the blocking of the UI is done by the Payment Request dialog itself. The Blocks API doesn't seem to offer a good solution for that yet. Ref: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4159

I guess once they fix that, the loading indicators should be hooked to some point in the [checkout lifecycle](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/checkout-flow-and-events.md#tracking-flow-through-status), so we could probably still take advantage of that while using the AJAX APIs.

I can't remember adding any loading indicators or any UI improvements manually by using the AJAX APIs in WCPay.

Is this something we did in Stripe? Would there be a better outcome if we used the Blocks APIs?

Reply:
> By "responsive" here, I ended up interpreting it as responsive in the sense of being adjustable to different screen sizes. But that's not the case, right?

Correct, I meant that as in "reacts to user actions", but I fully understand the confusion here 🤔 

I changed this to say "We have to manually make the UI react to user input" in a8bffe96ae96b23b548659d4f47f082aa6efabbc, let me know what you think of that change 🙏 

> Actually, the blocking of the UI and the loading indicators were missing from the Blocks implementation too.

Oh, I didn't know that! If they do add that sort of feature we don't have access to it though, so I think the point still stands 🤔 

I changed this to say "This refers to current and future improvements to the UX; things like...." in 4c1c385e51eaf2de81f9be2d57d552daccfea94a. Let me know what you think!

> I can't remember adding any loading indicators or any UI improvements manually by using the AJAX APIs in WCPay.
>
> Is this something we did in Stripe? Would there be a better outcome if we used the Blocks APIs?

No, we did not add loading indicators in Stripe; we deemed it too much effort for a small improvement in UX.

The Blocks do this automatically for credit card payments; I thought they did this for PRBs as well, but as you just pointed out that's not the case 🙂 

What I'm trying to get across with this point is that there are many UX improvements in the Blocks which we don't have access to, and there will probably be more improvements made over time that we _also_ won't have access to.

Let me know if you think the new phrasing gets that point across, or if you think I need to work on this a bit more 🤔